lib/view_component/test_helpers.rb

```diff
@@ -34,20 +36,22 @@ def render_inline(component, **args, &block)
       Nokogiri::HTML.fragment(@rendered_component)
     end
 
-    def controller
-      @controller ||= Base.test_controller.constantize.new.tap { |c| c.request = request }.extend(Rails.application.routes.url_helpers)
-    end
-
     def request
       @request ||= ActionDispatch::TestRequest.create
     end
 
-    def with_variant(variant)
-      old_variants = controller.view_context.lookup_context.variants
+    def with_variant(new_variant)
+      old_variant = variant
 
-      controller.view_context.lookup_context.variants = variant
+      @variant = new_variant
       yield
-      controller.view_context.lookup_context.variants = old_variants
+      @variant = old_variant
```

Comment:
Are there strong tests on this change? Its subtly different than the previous implementation in that the "old" version would be read from the controller lookup_context (which I assume could have other defaults etc)

Reply:
as an aside reading this code again I'm sort of confused why we have this pattern. Its strikes me it would be much simpler to pass `variant` as an argument to `render_inline` vs this wrapper. @joelhawksley any thoughts on why the API is like this?

Reply:
At some point during my work here, I did run into some test failures due to me not correctly modifying `controller.view_context.lookup_context.variants`, but I'm going to defer to the maintainers here for sure! If you think there's some test coverage missing I'm happy to add some.